### PR TITLE
solution

### PR DESCRIPTION
**Pull request by lytvyshko**

### CODE REVIEW

File: src/sortStudents.ts [hunk 1]
```diff
@@ -1,16 +1,47 @@
...
+function getAverageGrade(grades: number[]): number {
+  return grades.reduce((accum, grade) => accum + grade, 0) / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const studentsCopy = students.slice();
```
Comment (AndriiRut) [inline src/sortStudents.ts:28]: You can use `Spread operator` to make a copy of `objects` or `arrays`

**Review by AndriiRut — CHANGES_REQUESTED** [review]
Good Job! (=

File: src/sortStudents.ts [hunk 2]
```diff
@@ -1,16 +1,47 @@
...
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const studentsCopy = [...students];
+
+  if (sortBy === SortType.Name || sortBy === SortType.Surname) {
```
Comment (Anna-Tkachenko) [inline src/sortStudents.ts:30]: Check point 1 in checklist

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]

File: src/sortStudents.ts [hunk 3]
```diff
@@ -1,16 +1,49 @@
...
+        ? studentsCopy.sort((a, b) => getAverageGrade(a[sortBy])
+        - getAverageGrade(b[sortBy]))
+        : studentsCopy.sort((a, b) => getAverageGrade(b[sortBy])
+        - getAverageGrade(a[sortBy]));
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+    default:
+      return order === 'asc'
+        ? studentsCopy.sort((a, b) => +a[sortBy] - +b[sortBy])
```
Comment (Anna-Tkachenko) [inline src/sortStudents.ts:46]: Will be better to move all logic inside the sort method
And use default case for unexpected initial values or for throwing erros
This logic should be for 'born' and 'died' fields

**Review by Anna-Tkachenko — APPROVED** [review]

File: src/sortStudents.ts (hunk 3 above)
Comment (lytvyshko) [reply src/sortStudents.ts]: Have I understood you right?

File: src/sortStudents.ts [hunk 4]
```diff
@@ -1,16 +1,57 @@
...
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const studentsCopy = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      return studentsCopy.sort((a, b) => {
```
Comment (Anna-Tkachenko) [inline src/sortStudents.ts:33]: Will be better to write all logic inside method sort
If you'll want to change variable name for students array or change some variables inside method sort you should do it 3 times in your code

**Review by Anna-Tkachenko — APPROVED** [review]

**Pull request closed without merging**